python/tk_desktop/desktop_window.py

```diff
@@ -1359,11 +1359,37 @@ def __launch_app_proxy_for_project(
             self.ui.actionRefresh_Projects.setVisible(False)
 
             self.current_project = project
+            self.requested_pipeline_configuration_id = (
+                requested_pipeline_configuration_id
+            )
 
-            # trigger an update to the model to track this project access
-            self.__set_project_just_accessed(project)
-            QtGui.QApplication.instance().processEvents()
+            self.project_overlay.start_progress()
+            # Trigger an update to the model to track this project access
+            # this is done in a separate thread to avoid blocking the UI.
+            # After it finishes, we start phases 2 and 3 in __get_pipeline_configurations method
+            self.set_just_accessed_thread = StartAppProxyForProject(
+                self.__set_project_just_accessed, project
+            )
+            self.set_just_accessed_thread.finished.connect(
+                self.__get_pipeline_configurations
+            )
+            self.set_just_accessed_thread.start()
+        except Exception as error:
+            log.exception(str(error))
+            message = (
+                "%s"
+                "\n\nThere was an error connecting to Flow Production Tracking."
+                "\n\nFor more details, see the console." % str(error)
+            )
+            self.project_overlay.show_error_message(message)
+            return
 
+    def __get_pipeline_configurations(self):
```

Comment:
I'm not sure I like the name of this method. Any reason for it?

Why not something like `__launch_app_proxy_for_project_follow_up`?